File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/FinancialCores/Summit.cs

using Psi.Data.Models.ClientConfigurationModels.FinancialCores.SummitSettings;

namespace Psi.Data.Models.ClientConfigurationModels.FinancialCores
{
    public class Summit : SettingsBaseHelper
    {
        private RegularAchTransfers _regularAchTransfers;

        public Summit(ISettingsBase settingsBase) : base(settingsBase){}

        [SettingKey("FinancialCore.Summit.TransferCommandCode")]
        public string TransferCommandCode
		{
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("FinancialCore.Summit.UserFields.SegmintMarketingIdEnabled")]
        public string SegmintMarketingIdEnabled
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("FinancialCore.Summit.UseTwelveDigitTransactionAmount")]
        public bool UseTwelveDigitTransactionAmount
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        public RegularAchTransfers RegularAchTransfers
		{
            get => _regularAchTransfers ?? (_regularAchTransfers = new RegularAchTransfers(SettingsBase));
            set => _regularAchTransfers = value;
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/FinancialCores/Summit.ts
// Generated imports
import { RegularAchTransfers } from '../RegularAchTransfers';

export interface Summit {
    /** @settingKey FinancialCore.Summit.TransferCommandCode */
    transferCommandCode: string;
    /** @settingKey FinancialCore.Summit.UserFields.SegmintMarketingIdEnabled */
    segmintMarketingIdEnabled: string;
    /** @settingKey FinancialCore.Summit.UseTwelveDigitTransactionAmount */
    useTwelveDigitTransactionAmount: boolean;
    regularAchTransfers: RegularAchTransfers;
}
